cpp/src/fil/fil.cu

```diff
@@ -311,10 +334,12 @@ struct forest {
 
   virtual void free(const raft::handle_t& h)
   {
-    if (vector_leaf_len_ > 0) {
-      h.get_device_allocator()->deallocate(
-        vector_leaf_, sizeof(float) * vector_leaf_len_, h.get_stream());
-    }
+    auto deallocate = [&](void* ptr, size_t size) {
```

Comment:
Please use a named function for this, defined at the namespace level. If necessary, it can be later moved to raft.